control-userlayer/src/main/java/org/oskari/control/userlayer/CreateUserLayerHandler.java

```diff
@@ -252,13 +278,18 @@ private void checkZipContainsExactlyOneMainFile(Set<String> extensions) throws A
     private SimpleFeatureCollection parseFeatures(FileItem zipFile,
             Charset cs, Set<String> validFiles,
             CoordinateReferenceSystem sourceCRS,
-            CoordinateReferenceSystem targetCRS) throws ActionException {
+            CoordinateReferenceSystem targetCRS) throws ActionParamsException {
         File dir = null;
+        FeatureCollectionParser parser = null;
         try {
             dir = makeRandomTempDirectory();
             File mainFile = unZip(zipFile, cs, validFiles, dir);
-            FeatureCollectionParser parser = getParser(mainFile);
+            parser = getParser(mainFile);
             return parse(parser, mainFile, sourceCRS, targetCRS);
+        } catch (ServiceException e){
```

Comment:
This catch block is catching potential exceptions from makeRandomTempDirectory() and unZip() that have nothing to do with `"parser_error"`s. Should add a nested try block that calls `parser.parse()` and catch that ServiceException.